app/controllers/general.php

```diff
@@ -408,7 +408,7 @@ function router(App $utopia, Database $dbForConsole, SwooleRequest $swooleReques
         * @see https://www.owasp.org/index.php/List_of_useful_HTTP_headers
         */
         if (App::getEnv('_APP_OPTIONS_FORCE_HTTPS', 'disabled') === 'enabled') { // Force HTTPS
-            if ($request->getProtocol() !== 'https' && ($requestHeaders['host'] ?? '') !== 'localhost' && ($requestHeaders['host'] ?? '') !== APP_HOSTNAME_INTERNAL) { // localhost allowed for proxy, APP_HOSTNAME_INTERNAL allowed for migrations
+            if ($request->getProtocol() !== 'https' && ($swooleRequest->header['host'] ?? '') !== 'localhost' && ($swooleRequest->header['host'] ?? '') !== APP_HOSTNAME_INTERNAL) { // localhost allowed for proxy, APP_HOSTNAME_INTERNAL allowed for migrations
```

Comment:
Lets keep this to `$requestHeaders` itself since the issue with empty cookies is not taken care of

Reply:
$requestHeaders is defined elsewhere. It shouldn't be used here.

Reply:
![Screenshot 2023-10-31 at 11 07 53 PM](https://github.com/appwrite/appwrite/assets/20852629/de0b4f35-9dae-40d4-8fdc-55db9513b8c2)

This was the original PR that affected this line of code.. so requestHeaders is something we declared.

Reply:
$requestHeadres is defined in the `router()` function:

https://github.com/appwrite/appwrite/blob/9fded16d3f90d3a0be61350bcbb73905611cf471/app/controllers/general.php#L48-L197

not in the block i updated.

Reply:
Also, I don't think the cookie change is related to the line I updated.

Reply:
Got it 👍